### PR TITLE
Enable external CCM for Opentstack on cluster creation when supported

### LINKED ISSUE #4730: Openstack: move to out-of-tree cloud provider
**User Story**
As a part of out effort to get out out of in-tree cloud providers, we will start with switching Openstack to https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager.md

**Acceptance criteria**
- [ ] For Openstack cluster, we no longer configure the built-in cloud provider in new clusters
- [ ] Instead, we deploy https://github.com/kubernetes/cloud-provider-openstack
- [ ] If possible, all existing clusters are migrated from using the built-in to out-of-tree without destroing volumes or LBs
  - [ ] If that is not possible, then in-tree  provider configuration is preserved for existing clusters and they are marked to prevent breakage through auto-migration

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Enables Opentstack external CCM by default on creation of 1.16+ clusters.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4730 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Openstack: New Kubernetes 1.16+ clusters use the external Cloud Controller Manager and CSI by default
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -103,6 +105,13 @@ type ClusterSpec struct {
 	AuditLogging *AuditLoggingSettings `json:"auditLogging,omitempty"`
 }
 
+const (
+	// ClusterFeatureExternalCloudProvider describes the external cloud provider feature. It is
+	// only supported on a limited set of providers for a specific set of Kube versions. It must
+	// not be set if its not supported.
+	ClusterFeatureExternalCloudProvider = "externalCloudProvider"
+)
+
```
Comment (xrstf) [inline api/pkg/crd/kubermatic/v1/cluster.go:114]: For the operator I put all feature flags from all packages into https://github.com/kubermatic/kubermatic/blob/master/api/pkg/features/features.go -- maybe this one should go there, too?

Comment (xrstf) [reply api/pkg/crd/kubermatic/v1/cluster.go:114]: No idea why I forgot to move the existing constant for this feature over to the features.go as well :-D

Comment (alvaroaleman) [reply api/pkg/crd/kubermatic/v1/cluster.go:114]: Well I put it there because its part of the CRD which AFAIK is not the case for the other feature flags

**Pull request merged**